apparmor.d/groups/full-policy/full-policy

```diff
@@ -0,0 +1,251 @@
+  deny unix (setopt),
+  deny unix (getopt),
+  
+  ## won't break with the current set of profiles
+  deny ptrace (trace),
+  deny ptrace (tracedby),
+  deny ptrace (readby),
+  
+  ## won't break with the current set of profiles
+  deny network bluetooth,
+  deny network alg,
+  deny network ash,
+  deny network rose,
+  deny network x25,
+  deny network ax25,
+  deny network ipx,
+  deny network netrom,
+  deny network appletalk,
+  deny network econet,
+  deny network qipcrtr,
+  deny network bridge,
+  deny network atmpvc,
+  deny network netbeui,
+  deny network security,
+  deny network key,
+  deny network atmsvc,
+  deny network rds,
+  deny network irda,
+  deny network pppox,
+  deny network wanpipe,
+  deny network ib,
+  deny network mpls,
+  deny network can,
+  deny network tipc,
+  deny network rxrpc,
+  deny network isdn,
+  deny network phonet,
+  deny network ieee802154,
+  deny network caif,
+  deny network vsock,
+  deny network kcm,
+  deny network smc,
+  deny network xdp,
+  
+  # -----
+  
+  ## Section 2 - File permissions
+  
+  ## This is quite restrictive for a "general" profile.
+  ## Can of course be further restricted. Probably by a lot.
+  
+  ## The owner can read pretty much everything
+  ## He can also write to the directories
+  ## directly under root.
+  / r,
+  owner / rwlk,
+  owner /** rw,
+
+  ## Everyone can see the home directories
+  ## Only the owners allowed inside
+  /home r,
+  owner /home/** rwlkPix,
+
+  ## Reserved for the owner 'root' only
+  owner /boot/** rwlk,
+  owner /root/** rwlk,
+
+  ## Running stuff is fine here
+  ## Modifying requires ownership
+  @{lib}/** rPix,
+  owner @{lib}/** rwmlkPix,
+  # Here too
+  @{bin}/** rPix,
+  owner @{bin}/** rwmlkPix,
+  # And here
+  /opt/** rPix,
+  owner /opt/** rwmlkPix,
+  
+  ## Owner can modify only
+  /usr/** r,
+  owner /usr/** rwlk,
+
+  ## Don't you dare read others' temp files
+  owner /{,var/}tmp/** rw,
+ 
+  ## Reading etc is cool
+  ## No modification allowed if not the owner
+  /{,usr/local/}etc/** r,
+  owner /{,usr/local/}etc/** rwmlk,
+
+  ## Can be restricted? Maybe
+  /dev/** rw,
+
+  ## Owner can access his media
+  owner /media/** rw,
+  
+  ## Or what is mounted
+  owner /mnt/** rw,
+
+  ## Restricting this a little harder
+  ## at least we preven executing the temp and logs
+  /var/** rwmlkPix,
+  deny /var/log/** x,
+  deny /var/tmp/** x,
+
+  ## Can be restricted
+  @{run}/** rw,
+  owner @{run}/** rwlk,
+
+  ## Reading can be more restricted for subdirs
+  @{PROC}/** r,
+  owner @{PROC}/** rw,
+  
+  ## Can definetely be restricted further
+  @{sys}/** rw,
+
+  ## Deny changing the profile files
+  deny /**/apparmor.d/** w,
+  
+  ## No access to memory and por
+  deny /dev/mem rw,
+  deny /dev/kmem rw,
+  
+  ## No access to I/O ports
+  deny /dev/port rw,
+  
+  ## This blocks what Kicksecure security-misc package blocks.
+  deny /**System.map* rw,
+  
+  ## No accessing the disk in other ways to circumvent the policy
```

Comment:
Note sure about this. systemd really does not need disks access? (not even mentionning no-new-privs, in transioned profiles).

Reply:
My understanding is, only select services do, which have their respective profiles. Because this caused no breakage or slowdown.

Reply:
I think you should test it again. With `earlypolicy`, it prevents udevd to start, and the boot cannot finish.

Reply:
Ooof. I see. Will fix.

Reply:
I don't understand why systemd-udevd tries to run under full-policy eventhough we have a dedicated profile [for it](https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/groups/systemd/systemd-udevd). I think this migght be the cause of the issue. Is this a bug, or is it about how we load the profile? The installer seems to skip this profile for some reason.

Reply:
Without `earlypolicy`, this is because the old initframfs hooks only load the profile for systemd. So at very early stage (when udevd is started) there is no profile for udevd yet. `earlypolicy` fix it as all profiles are available from cache from the very moment systemd starts.

Then, it move to the second issue. It is a due to a no_new_privs error. If you look at the apparmor log (`aa-log -s | grep error`) you should see that it cannot transition to some profile due to no_new_privs, and therefore the transition fails and the program runs inside the parent profile. As the parent profile deny disk access, the boot it broken.